Build error with clang in case of minimal PSA crypto config

### Summary

There is a build error when Mbed TLS is build with PSA Crypto support with a minimal configuration. clang reports unreachable code in `library/psa_crypto_aead.c`, `library/psa_crypto_cipher.c` and `library/psa_crypto.c`. When all the algorithms are disabled, some of the functions return immediately, and some code at the end of the functions become unreachable. 

Please note that the attached reproducer won't trigger the errors in `library/psa_crypto.c`. I'm not sure why.

### System information

Mbed TLS version: v3.4.0 and developement branch: 
Operating system and version: Bare metal and Linux
Configuration (if not default, please attach `mbedtls_config.h`): [my_config.zip](https://github.com/Mbed-TLS/mbedtls/files/11520191/my_config.zip)
Compiler and options: cmake with clang 13.0.1. See steps to reproduce
Additional environment information: -

### Expected behavior
Mbed TLS builds without error

### Actual behavior
Compilation fails with error


### Steps to reproduce
```
rm -fr build && mkdir build && pushd build && \
cmake -DCMAKE_C_COMPILER=clang .. && \
cmake --build . ; \
popd
```

### Additional information
```
/home2/mate/temp/mbedtls/library/psa_crypto_aead.c:126:27: error: code will never be executed [-Werror,-Wunreachable-code]
    operation->key_type = psa_get_key_type(attributes);
                          ^~~~~~~~~~~~~~~~
/home2/mate/temp/mbedtls/library/psa_crypto_aead.c:212:9: error: code will never be executed [-Werror,-Wunreachable-code]
    if (status == PSA_SUCCESS) {
        ^~~~~~
/home2/mate/temp/mbedtls/library/psa_crypto_aead.c:322:9: error: code will never be executed [-Werror,-Wunreachable-code]
    if (status == PSA_SUCCESS) {
        ^~~~~~
/home2/mate/temp/mbedtls/library/psa_crypto_aead.c:557:9: error: code will never be executed [-Werror,-Wunreachable-code]
    if (status == PSA_SUCCESS) {
        ^~~~~~
/home2/mate/temp/mbedtls/library/psa_crypto_aead.c:628:9: error: code will never be executed [-Werror,-Wunreachable-code]
    if (status == PSA_SUCCESS) {
        ^~~~~~
5 errors generated.
library/CMakeFiles/mbedcrypto.dir/build.make:899: recipe for target 'library/CMakeFiles/mbedcrypto.dir/psa_crypto_aead.c.o' failed
make[2]: *** [library/CMakeFiles/mbedcrypto.dir/psa_crypto_aead.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/home2/mate/temp/mbedtls/library/psa_crypto_cipher.c:154:9: error: code will never be executed [-Werror,-Wunreachable-code]
    if (cipher_id != NULL) {
        ^~~~~~~~~
1 error generated.
```

```
mbedtls/library/psa_crypto.c:7945:14: error: code will never be executed [-Werror,-Wunreachable-code]
    status = psa_driver_wrapper_pake_input(operation, driver_step,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
mbedtls/library/psa_crypto.c:8002:14: error: code will never be executed [-Werror,-Wunreachable-code]
    status = psa_driver_wrapper_pake_get_implicit_key(operation,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/ChangeLog.d/fix-empty-enum.txt
+++ b/ChangeLog.d/fix-empty-enum.txt
@@ -1,3 +1,3 @@
 Bugfix
-   * Fix compile failure due to empty enum in cipher_wrap.o, when building
+   * Fix compile failure due to empty enum in cipher_wrap.c, when building
      with a very minimal configuration. Fixes #7625.

--- a/ChangeLog.d/fix-empty-enum.txt
+++ b/ChangeLog.d/fix-empty-enum.txt
@@ -1,0 +1,3 @@
+Bugfix
+   * Fix compile failure due to empty enum in cipher_wrap.o, when building
+     with a very minimal configuration. Fixes #7625.

--- a/library/cipher_wrap.c
+++ b/library/cipher_wrap.c
@@ -120,8 +120,10 @@ enum mbedtls_cipher_base_index {
     MBEDTLS_CIPHER_BASE_INDEX_NULL_BASE,
 #endif
 #if defined(MBEDTLS_CIPHER_MODE_XTS) && defined(MBEDTLS_AES_C)
-    MBEDTLS_CIPHER_BASE_INDEX_XTS_AES
+    MBEDTLS_CIPHER_BASE_INDEX_XTS_AES,
 #endif
+    /* Prevent compile failure due to empty enum */
+    MBEDTLS_PREVENT_EMPTY_ENUM
 };
 
 #if defined(MBEDTLS_GCM_C)

--- a/library/cipher_wrap.c
+++ b/library/cipher_wrap.c
@@ -123,7 +123,7 @@ enum mbedtls_cipher_base_index {
     MBEDTLS_CIPHER_BASE_INDEX_XTS_AES,
 #endif
     /* Prevent compile failure due to empty enum */
-    MBEDTLS_PREVENT_EMPTY_ENUM
+    MBEDTLS_CIPHER_BASE_PREVENT_EMPTY_ENUM
 };
 
 #if defined(MBEDTLS_GCM_C)

